tests/ui/traits/trivial_impl4.rs

```diff
@@ -0,0 +1,22 @@
+//! Check that we don't break orphan rules.
+//! Check that we don't break orphan rules.
+//! The dependency may add an impl for `u8` later,
+//! which would break this crate. We want to avoid adding
+//! more ways in which adding an impl can be a breaking change.
+//! This test differs from `trivial_impl4` because there actually
+//! exists any impl for `Trait`, which has an effect on coherence.
```

Comment:
? In what way does this affect things. Coherence is not allowed to assume that a trait is never implemented just because it cant see any impls in dependencies or locally

Reply:
I know, and I want to make sure it doesn't with this test 😅

Reply:
I know, and I want to make sure it doesn't with this test 😅